bpf/lib/nodeport.h

```diff
@@ -503,7 +515,8 @@ static __always_inline int nodeport_lb6(struct __ctx_buff *ctx,
 		return DROP_INVALID;
 
 	backend_local = lookup_ip6_endpoint(ip6);
-	if (lb6_svc_is_hostport(svc) && !backend_local)
+	if (!backend_local && (lb6_svc_is_hostport(svc) ||
+			       nodeport_must_drop_remote()))
```

Comment:
Is this valid for the path here that is sourced from the tunnel? Or there can we safely assume that the traffic was LB'd to the backend already at the source node, even if the target nodeport address is an intermediate node?

Reply:
Good catch, for the tunnel we need to handle it in tc ingress as well, will fix.